sdl-core/src/main/scala/io/smartdatalake/workflow/dataframe/spark/SparkDataFrame.scala

```diff
@@ -0,0 +1,269 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2022 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.dataframe.spark
+
+import io.smartdatalake.config.SdlConfigObject.DataObjectId
+import io.smartdatalake.definitions.Environment
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.util.misc.SchemaUtil
+import io.smartdatalake.util.spark.DataFrameUtil
+import io.smartdatalake.workflow.ActionPipelineContext
+import io.smartdatalake.workflow.dataframe._
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Column, DataFrame, RelationalGroupedDataset, Row}
+
+import scala.reflect.runtime.universe
+import scala.reflect.runtime.universe.typeOf
+
+case class SparkDataFrame(inner: DataFrame) extends GenericDataFrame {
+  override def subFeedType: universe.Type = typeOf[SparkSubFeed]
+  override def schema: SparkSchema = SparkSchema(inner.schema)
+  override def join(other: GenericDataFrame, joinCols: Seq[String]): SparkDataFrame = {
```

Comment:
One additional use-case of generic transformations that comes to mind is Unit-Testing:
As Snowflake does not allow you to do proper Unit-Tests with connecting to the cloud, using generic functions would be a sort of workaround to allow you to to switch execution engines for Unit-Testing. 
So for that, the user would need to replace the SnowflakeTableDataObject that he uses in production with for example HiveTableDataObjects in the unit-tests and therefore SDLB would automatically switch to Spark during Unit-Tests.
Not sure how realistic this is given that having a different engine for your tests may make the code behave differently for tests... It might be worth a shot to try it out when migrating a SDL project from Spark to Snowpark.
For SDL-Internal transformations we already do that implicitly by using Spark as Engine all the time when there is no SnowflakeTableDataObject involved.